app/helpers/generic_file_helper.rb

```diff
@@ -21,8 +21,9 @@ def render_download_link text = nil
     link_to (text || "Download"), download_path(@generic_file), { id: "file_download", target: "_new", data: { label: @generic_file.id } }
   end
 
-  def download_path gf
-    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id)
+  # sufia.download path is from ActionDispatch::Routing::PolymorphicRoutes#polymorphic_path
+  def download_path gf, options = {}
+    GenericFile.find(gf.id).doi_url || sufia.download_path(gf.id, options)
```

Comment:
Few comments (I dug into this in my own branch because it bit me, too):

```ruby
def download_path gf, options = {}
```

This still won't work in all cases, because you're assuming download_path takes at most 2 arguments, the second of which must be a hash. In actuality:

```ruby
irb(main):001:0> Sufia::Engine.routes.url_helpers.method(:download_path).arity
=> -1
```
-1 indicates that the method has variable arity, so you need to match that definition:

```ruby
def download_path(gf, *more)
....
.... || sufia.download_path(gf.id, *more)
```

```ruby
GenericFile.find(gf.id).doi_url || ...
```
I see a couple of questions/issues here:

1) This is called above as

```ruby
download_path(@generic_file)
```
so gf is a GenericFile which means

```ruby
002:0> gf == GenericFile.find(gf.id) 
=> true
```
Is there any point to this find call? If there is, we should add a comment explaining the logic behind it, because at first glance it's redundant.

2) doi_url.present? will work better than just .doi_url if doi_url is ever an empty string

3) Is the assumption that everything passed into download_path will respond to doi_url safe? Sufia doesn't make this assumption in it's download_path code, so I'm a bit leary of it. Might be better to check with respond_to? before doing this.